test/Microsoft.TemplateEngine.Cli.UnitTests/TemplateResolutionTests/ListTemplateListResolverTests.cs

```diff
@@ -414,6 +412,63 @@ public void TestGetTemplateResolutionResult_AuthorMatch(string templateAuthor, s
             Assert.False(matchResult.HasBaselineMismatch);
         }
 
+        [Theory(DisplayName = nameof(TestGetTemplateResolutionResult_TagsMatch))]
+        [InlineData("TestTag", "TestTag", true)]
+        [InlineData("Tag1||Tag2", "Tag1", true)]
+        [InlineData("Tag1||Tag2", "Tag", false)]
+        [InlineData("", "Tag", false)]
+        [InlineData("TestTag", "Other", false)]
+        [InlineData("TestTag", "", true)]
+        [InlineData("TestTag", null, true)]
+        [InlineData("TestTag", "TeSTTag", true)]
+        [InlineData("TestTag", "TeştTag", false)]
+        [InlineData("match_middle_test", "middle", false)]
+        [InlineData("input", "İnput", false)]
+        public void TestGetTemplateResolutionResult_TagsMatch(string templateTags, string commandTag, bool matchExpected)
+        {
+            const string separator = "||";
+            string[] templateTagsArray = templateTags.Split(separator);
+
+            List<ITemplateInfo> templatesToSearch = new List<ITemplateInfo>();
+
+            templatesToSearch.Add(
+               new MockTemplateInfo("console", name: "Long name for Console App", identity: "Console.App.T1", groupIdentity: "Console.App.Test", author: "TemplateAuthor")
+                   .WithTag("language", "L1")
+                   .WithTag("type", "project")
+                   .WithClassifications(templateTagsArray)
```

Comment:
Stupid question: Are we calling the right method here? `Classifications` doesn't sound like it's related to `tags`.

Reply:
`Classifications` are named `Tags` in UX, I now that's confusing. I think we should do some renames\restructure in the code to have everything named unambiguously.